--- pyclewn_install.py	2011-11-23 21:14:52.000000000 +0100
+++ pyclewn_install.py.new	2011-11-23 21:19:31.000000000 +0100
@@ -59,8 +59,6 @@
 def build_vimhelp():
     """Add pyclewn help to Vim help."""
     helpdir = pathjoin(vimdir(), 'doc')
-    print >> sys.stderr, 'running Vim help tags file generation in %s' % helpdir
-    vim.exec_vimcmd(['helptags ' + helpdir, 'echo v:version'])
 
 def unlink(filename):
     """Delete a file."""
