diff --git a/10x20.bdf b/10x20.bdf
index c6a26a0..e7c02c8 100644
--- a/10x20.bdf
+++ b/10x20.bdf
@@ -502,9 +502,9 @@ BITMAP
 3300
 3300
 6180
-6180
-6180
-6180
+6380
+6D80
+7180
 6180
 3300
 3300
diff --git a/12x13ja.bdf b/12x13ja.bdf
index cd33cab..20c4e20 100644
--- a/12x13ja.bdf
+++ b/12x13ja.bdf
@@ -385,9 +385,9 @@ BITMAP
 0E00
 3180
 60C0
-60C0
-60C0
-60C0
+63C0
+64C0
+78C0
 60C0
 3180
 0E00
diff --git a/18x18ja.bdf b/18x18ja.bdf
index 3644514..a5f4528 100644
--- a/18x18ja.bdf
+++ b/18x18ja.bdf
@@ -469,12 +469,12 @@ BITMAP
 021000
 040800
 040800
-0C0C00
-080400
-080400
-080400
-080400
-0C0C00
+0C1C00
+082400
+084400
+088400
+090400
+0E0C00
 040800
 040800
 021000
diff --git a/18x18ko.bdf b/18x18ko.bdf
index c79dbf3..987ee65 100644
--- a/18x18ko.bdf
+++ b/18x18ko.bdf
@@ -474,10 +474,10 @@ BITMAP
 040800
 040800
 0C0C00
-080400
-080400
-080400
-080400
+083400
+084400
+088400
+0B0400
 0C0C00
 040800
 040800
diff --git a/5x7.bdf b/5x7.bdf
index 06cde34..0683371 100644
--- a/5x7.bdf
+++ b/5x7.bdf
@@ -274,12 +274,12 @@ SWIDTH 685 0
 DWIDTH 5 0
 BBX 5 7 0 -1
 BITMAP
-20
-50
-50
-50
-50
-20
+60
+90
+B0
+D0
+90
+60
 00
 ENDCHAR
 STARTCHAR one
diff --git a/5x8.bdf b/5x8.bdf
index 50637b4..4149fa7 100644
--- a/5x8.bdf
+++ b/5x8.bdf
@@ -291,12 +291,12 @@ DWIDTH 5 0
 BBX 5 8 0 -1
 BITMAP
 00
-20
-50
-50
-50
-50
-20
+60
+90
+B0
+D0
+90
+60
 00
 ENDCHAR
 STARTCHAR one
diff --git a/6x10.bdf b/6x10.bdf
index c03715f..723c969 100644
--- a/6x10.bdf
+++ b/6x10.bdf
@@ -328,7 +328,7 @@ BITMAP
 20
 50
 88
-88
+A8
 88
 50
 20
diff --git a/6x12.bdf b/6x12.bdf
index 3fbf113..5907922 100644
--- a/6x12.bdf
+++ b/6x12.bdf
@@ -363,8 +363,8 @@ BITMAP
 00
 30
 48
-48
-48
+58
+68
 48
 48
 30
diff --git a/6x13.bdf b/6x13.bdf
index 921a15e..76c3866 100644
--- a/6x13.bdf
+++ b/6x13.bdf
@@ -380,9 +380,9 @@ BITMAP
 20
 50
 88
-88
-88
-88
+98
+A8
+C8
 88
 50
 20
diff --git a/6x13B.bdf b/6x13B.bdf
index a01b80e..dc7cb62 100644
--- a/6x13B.bdf
+++ b/6x13B.bdf
@@ -380,8 +380,8 @@ BITMAP
 30
 78
 CC
-CC
-CC
+DC
+EC
 CC
 CC
 78
diff --git a/6x13O.bdf b/6x13O.bdf
index 59d4fd3..f6b7830 100644
--- a/6x13O.bdf
+++ b/6x13O.bdf
@@ -380,9 +380,9 @@ BITMAP
 10
 28
 44
-44
-44
-88
+4C
+B4
+C8
 88
 50
 20
diff --git a/6x9.bdf b/6x9.bdf
index 4ae8eae..e1a6476 100644
--- a/6x9.bdf
+++ b/6x9.bdf
@@ -310,8 +310,8 @@ BITMAP
 00
 30
 48
-48
-48
+58
+68
 48
 30
 00
diff --git a/7x13.bdf b/7x13.bdf
index 07db3c6..55864c8 100644
--- a/7x13.bdf
+++ b/7x13.bdf
@@ -381,9 +381,9 @@ BITMAP
 30
 48
 84
-84
-84
-84
+8C
+B4
+C4
 84
 48
 30
diff --git a/7x13B.bdf b/7x13B.bdf
index 2ef2a38..9ab351c 100644
--- a/7x13B.bdf
+++ b/7x13B.bdf
@@ -381,8 +381,8 @@ BITMAP
 30
 48
 CC
-CC
-CC
+DC
+EC
 CC
 CC
 48
diff --git a/7x13O.bdf b/7x13O.bdf
index 4deec2e..b6e7825 100644
--- a/7x13O.bdf
+++ b/7x13O.bdf
@@ -381,9 +381,9 @@ BITMAP
 18
 24
 42
-42
-42
-84
+4E
+92
+E4
 84
 48
 30
diff --git a/7x14.bdf b/7x14.bdf
index 5611c23..1ce528b 100644
--- a/7x14.bdf
+++ b/7x14.bdf
@@ -397,10 +397,10 @@ BITMAP
 30
 48
 84
-84
-84
-84
-84
+8C
+94
+A4
+C4
 84
 48
 30
diff --git a/7x14B.bdf b/7x14B.bdf
index d71994e..c3661c4 100644
--- a/7x14B.bdf
+++ b/7x14B.bdf
@@ -397,10 +397,10 @@ BITMAP
 78
 CC
 CC
-CC
-CC
-CC
-CC
+DC
+DC
+EC
+EC
 CC
 CC
 78
diff --git a/8x13.bdf b/8x13.bdf
index a95b718..f9136b0 100644
--- a/8x13.bdf
+++ b/8x13.bdf
@@ -380,9 +380,9 @@ BITMAP
 18
 24
 42
-42
-42
-42
+46
+5A
+62
 42
 24
 18
diff --git a/8x13B.bdf b/8x13B.bdf
index 0416c25..4a99d6e 100644
--- a/8x13B.bdf
+++ b/8x13B.bdf
@@ -379,10 +379,10 @@ BITMAP
 38
 6C
 C6
-C6
-C6
-C6
-C6
+CE
+DE
+F6
+E6
 C6
 6C
 38
diff --git a/8x13O.bdf b/8x13O.bdf
index 5715f2c..a8167f3 100644
--- a/8x13O.bdf
+++ b/8x13O.bdf
@@ -380,9 +380,9 @@ BITMAP
 18
 24
 42
-42
-42
-84
+4E
+92
+E4
 84
 48
 30
diff --git a/9x15.bdf b/9x15.bdf
index 68d9709..4b44b1c 100644
--- a/9x15.bdf
+++ b/9x15.bdf
@@ -414,10 +414,10 @@ BITMAP
 1C00
 2200
 4100
-4100
-4100
-4100
-4100
+4300
+4D00
+5100
+6100
 4100
 2200
 1C00
diff --git a/9x15B.bdf b/9x15B.bdf
index 2fa9a09..2b3d951 100644
--- a/9x15B.bdf
+++ b/9x15B.bdf
@@ -414,10 +414,10 @@ BITMAP
 1800
 3C00
 6600
-C300
-C300
-C300
-C300
+C700
+CB00
+D300
+E300
 6600
 3C00
 1800
diff --git a/9x18.bdf b/9x18.bdf
index f64c33f..9d65f38 100644
--- a/9x18.bdf
+++ b/9x18.bdf
@@ -467,10 +467,10 @@ BITMAP
 1C00
 2200
 4100
-4100
-4100
-4100
-4100
+4300
+4D00
+5100
+6100
 4100
 2200
 1C00
diff --git a/9x18B.bdf b/9x18B.bdf
index 6087c71..1786f31 100644
--- a/9x18B.bdf
+++ b/9x18B.bdf
@@ -467,10 +467,10 @@ BITMAP
 1C00
 3600
 6300
-6300
-6300
-6300
-6300
+6700
+6F00
+7B00
+7300
 6300
 3600
 1C00
