--- fs-uae-2.4.3_orig/src/fs-uae/menu.c	2015-01-27 21:48:55.000000000 +0100
+++ fs-uae-2.4.3/src/fs-uae/menu.c	2016-02-14 18:40:33.374505033 +0100
@@ -7,7 +7,7 @@
 #include <fs/time.h>
 #include "fs-uae.h"
 
-#define NUM_SAVE_SLOTS 9
+#define NUM_SAVE_SLOTS 17
 /// TRANSLATORS: This is a menu entry and must not be too long: If you struggle
 /// to get in short enough, just translate "No Device" instead.
 #define NO_HOST_DEVICE _("No Host Device")
