diff -ur urxvt-theme-switch_orig/color-themes urxvt-theme-switch/color-themes
--- urxvt-theme-switch_orig/color-themes	2022-10-06 20:03:34.863989585 +0200
+++ urxvt-theme-switch/color-themes	2022-10-08 11:52:21.193377761 +0200
@@ -38,7 +38,7 @@
     $self->{theme_dir}       = $path;
     $self->{auto_save_state} = $self->x_resource('color-themes.autosave') || 0;
     $self->{settingsfile} =
-      glob( $self->x_resource('color-themes.state-file') || "~/.urxvt-theme" );
+      glob( $self->x_resource('color-themes.state-file') || "~/.config/urxvt-theme" );
     $self->{preprocessor} =
       glob( $self->x_resource('color-themes.preprocessor') || "" );
 
