diff --git a/src/window.c b/src/window.c
index 1f16f458..a9bad2f6 100644
--- a/src/window.c
+++ b/src/window.c
@@ -1902,10 +1902,8 @@ void wWindowConstrainSize(WWindow *wwin, unsigned int *nwidth, unsigned int *nhe
 	int baseH = 0;
 
 	if (wwin->normal_hints) {
-		if (!wwin->flags.maximized) {
-			winc = wwin->normal_hints->width_inc;
-			hinc = wwin->normal_hints->height_inc;
-		}
+		winc = wwin->normal_hints->width_inc;
+		hinc = wwin->normal_hints->height_inc;
 		if (wwin->normal_hints->min_width > minW)
 			minW = wwin->normal_hints->min_width;
 		if (wwin->normal_hints->min_height > minH)
@@ -1980,17 +1978,15 @@ void wWindowConstrainSize(WWindow *wwin, unsigned int *nwidth, unsigned int *nhe
 		}
 	}
 
-	if (!wwin->flags.maximized) {
-		if (baseW != 0)
-			width = (((width - baseW) / winc) * winc) + baseW;
-		else
-			width = (((width - minW) / winc) * winc) + minW;
+	if (baseW != 0)
+		width = (((width - baseW) / winc) * winc) + baseW;
+	else
+		width = (((width - minW) / winc) * winc) + minW;
 
-		if (baseH != 0)
-			height = (((height - baseH) / hinc) * hinc) + baseH;
-		else
-			height = (((height - minH) / hinc) * hinc) + minH;
-	}
+	if (baseH != 0)
+		height = (((height - baseH) / hinc) * hinc) + baseH;
+	else
+		height = (((height - minH) / hinc) * hinc) + minH;
 
 	/* broken stupid apps may cause preposterous values for these.. */
 	if (width > 0)
