diff -ur klavaro-3.04_orig/gtkdatabox/Makefile.am klavaro-3.04/gtkdatabox/Makefile.am
--- klavaro-3.04_orig/gtkdatabox/Makefile.am	2019-02-02 21:04:48.000000000 +0100
+++ klavaro-3.04/gtkdatabox/Makefile.am	2021-05-31 10:11:00.705978936 +0200
@@ -1,6 +1,6 @@
 # From gtkdatabox git repo, adapted
 
-lib_LTLIBRARIES		= libgtkdataboks.la
+noinst_LTLIBRARIES		= libgtkdataboks.la
 libgtkdataboks_la_SOURCES= \
 			gtkdatabox.c\
 			gtkdatabox_marshal.c\
