diff -ur klavaro-3.04_orig/data/Makefile.am klavaro-3.04/data/Makefile.am
--- klavaro-3.04_orig/data/Makefile.am	2019-02-01 20:18:40.000000000 +0100
+++ klavaro-3.04/data/Makefile.am	2021-05-31 10:12:24.837980990 +0200
@@ -168,11 +168,11 @@
 EXTRA_DIST = $(klavaro_data)
 else
 @INTLTOOL_DESKTOP_RULE@
-desktopdir = $(prefix)/$(DATADIRNAME)/applications
+desktopdir = $(datadir)/$(DATADIRNAME)/applications
 desktop_in_file = klavaro.desktop.in
 desktop_DATA = $(desktop_in_file:.desktop.in=.desktop)
 @INTLTOOL_XML_RULE@
-appdatadir = $(prefix)/$(DATADIRNAME)/appdata
+appdatadir = $(datadir)/$(DATADIRNAME)/appdata
 appdata_in_files = klavaro.appdata.xml.in
 appdata_DATA = $(appdata_in_files:.xml.in=.xml)
 DISTCLEANFILES = $(desktop_DATA) \
diff -ur klavaro-3.04_orig/src/Makefile.am klavaro-3.04/src/Makefile.am
--- klavaro-3.04_orig/src/Makefile.am	2021-05-31 10:07:46.469974196 +0200
+++ klavaro-3.04/src/Makefile.am	2021-05-31 10:13:11.897982139 +0200
@@ -20,7 +20,7 @@
 
 AM_CPPFLAGS = @GTK_CFLAGS@ \
 	      -DPACKAGE_DATA_DIR=\""$(datadir)"\" \
-	      -DPACKAGE_LOCALE_DIR=\""$(localedir)"\"
+		  -DPACKAGE_LOCALE_DIR=\""$(datadir)/locale"\"
 
 AM_CFLAGS = -Wextra \
 	    -I$(top_srcdir)/gtkdatabox
