diff -ur klavaro-3.04_orig/src/Makefile.am klavaro-3.04/src/Makefile.am
--- klavaro-3.04_orig/src/Makefile.am	2019-01-31 22:55:16.000000000 +0100
+++ klavaro-3.04/src/Makefile.am	2021-05-31 10:07:46.469974196 +0200
@@ -29,7 +29,7 @@
 
 klavaro_LDADD = @GTK_LIBS@ \
 		-lgmodule-2.0 \
-	       	$(top_srcdir)/gtkdatabox/libgtkdataboks.la
+	       	$(top_builddir)/gtkdatabox/libgtkdataboks.la
 
 if IS_POSIX
 AM_CFLAGS += -export-dynamic
