diff -ur klavaro-3.04_orig/data/klavaro.desktop.in klavaro-3.04/data/klavaro.desktop.in
--- klavaro-3.04_orig/data/klavaro.desktop.in	2017-06-25 14:00:33.000000000 +0200
+++ klavaro-3.04/data/klavaro.desktop.in	2021-05-31 10:14:51.309984565 +0200
@@ -3,7 +3,7 @@
 _GenericName=Yet another touch typing tutor
 _Comment=A very flexible and efficient touch typing tutor.
 # Translators: these are keywords related to the program. CAI above means Computer Aided Instruction. Feel free to add and/or change the keywords in your language.
-_Keywords=keyboard;typing;tutor;CAI
+_Keywords=keyboard;typing;tutor;CAI;
 
 Exec=klavaro
 Icon=klavaro
