diff -ur a/configure.ac b/configure.ac
--- a/configure.ac	2023-02-26 16:55:55.682006975 +0100
+++ b/configure.ac	2023-02-26 16:56:13.722386073 +0100
@@ -202,7 +202,7 @@
 TAGLIB_DEPS="taglib_c >= 1.6.0"
 AS_IF([test "x$enable_mp4" != "xno"],
       [PKG_CHECK_EXISTS([$TAGLIB_DEPS], [have_taglib=yes], [have_taglib=no])],
-      [have_wavpack=no])
+      [have_taglib=no])
 
 AS_IF([test "x$have_taglib" != "xno"],
       [AC_DEFINE([ENABLE_MP4], [], [Define for taglib MP4 support])],
