diff -ur a/src/wavpack_tag.c b/src/wavpack_tag.c
--- a/src/wavpack_tag.c	2023-02-26 17:09:51.951580194 +0100
+++ b/src/wavpack_tag.c	2023-02-26 17:10:41.316617543 +0100
@@ -82,8 +82,6 @@
  */
 gboolean Wavpack_Tag_Read_File_Tag (gchar *filename, File_Tag *FileTag)
 {
-    if (!filename || !FileTag)
-        return FALSE;
 
     WavpackContext *wpc;
     gchar *field, *field2;
@@ -91,6 +89,9 @@
 
     int open_flags = OPEN_TAGS;
 
+    if (!filename || !FileTag)
+        return FALSE;
+
     wpc = WavpackOpenFileInput(filename, NULL, open_flags, 0);
 
     if ( wpc == NULL ) {
@@ -271,8 +272,6 @@
 
 gboolean Wavpack_Tag_Write_File_Tag (ET_File *ETFile)
 {
-    if (!ETFile || !ETFile->FileTag)
-        return FALSE;
 
     WavpackContext *wpc;
 
@@ -282,6 +281,9 @@
 
     int open_flags = OPEN_EDIT_TAGS;
 
+    if (!ETFile || !ETFile->FileTag)
+        return FALSE;
+
     wpc = WavpackOpenFileInput(filename, NULL, open_flags, 0);
 
     if ( wpc == NULL ) {
