diff -ur a/src/misc.c b/src/misc.c
--- a/src/misc.c	2013-02-05 23:42:37.000000000 +0100
+++ b/src/misc.c	2023-02-26 17:08:00.397236011 +0100
@@ -479,7 +479,7 @@
         /* Get the current date */
         memcpy(&t0, localtime(&t), sizeof(struct tm));
         /* Put the current year in 'current_year' tab */
-        sprintf(current_year,"%04d",1900+t0.tm_year);
+        snprintf(current_year,sizeof(current_year),"%04d",1900+t0.tm_year%1000u);
 
         tmp = &current_year[4-strlen(year)];
         if ( atoi(year) <= atoi(tmp) )
