diff -ur a/easytag.desktop.in b/easytag.desktop.in
--- a/easytag.desktop.in	2013-01-16 10:02:13.000000000 +0100
+++ b/easytag.desktop.in	2023-02-26 16:55:03.448909358 +0100
@@ -6,7 +6,7 @@
 Type=Application
 Icon=easytag
 Categories=GTK;AudioVideo;AudioVideoEditing;
-MimeType=x-directory/normal;inode/directory;audio/x-mp3;audio/x-mpeg;audio/mpeg;application/ogg;audio/x-vorbis+ogg;audio/x-flac;audio/x-musepack;audio/x-ape;
+MimeType=audio/x-mp3;audio/x-mpeg;audio/mpeg;application/ogg;audio/x-vorbis+ogg;audio/x-flac;audio/x-musepack;audio/x-ape;
 Exec=easytag %F
 Terminal=false
 StartupNotify=true
