diff -ur mc-4.7.5.3_orig/src/filemanager/midnight.c mc-4.7.5.3/src/filemanager/midnight.c
--- mc-4.7.5.3_orig/src/filemanager/midnight.c	2011-07-26 14:55:53.000000000 +0200
+++ mc-4.7.5.3/src/filemanager/midnight.c	2011-08-02 20:58:08.000000000 +0200
@@ -797,7 +797,7 @@
 
     tty_setup_sigwinch (sigwinch_handler);
 
-    if ((tty_baudrate () < 9600) || tty_is_slow ())
+    if ((tty_baudrate() > 0 && tty_baudrate() < 9600) || tty_is_slow())
         verbose = 0;
 
     init_xterm_support ();
