--- wmpower-0.4.3_orig/src/power_management/acpi/libacpi.c	2005-09-06 12:09:10.000000000 +0200
+++ wmpower-0.4.3/src/power_management/acpi/libacpi.c	2010-12-24 14:31:01.000000000 +0100
@@ -77,9 +77,6 @@
   struct dirent *batt;
   char *name;
 
-	/* do proc entries for acpi exist? */
-	if (access("/proc/acpi/info", R_OK) != 0) return 0;
-
   /* now enumerate batteries */
   batt_count = 0;
   battdir = opendir ("/proc/acpi/battery");
