--- sylpheed-2.7.1_orig/src/folderview.c	2009-06-15 06:35:28.000000000 +0200
+++ sylpheed-2.7.1/src/folderview.c	2010-07-23 21:53:07.000000000 +0200
@@ -1634,8 +1634,9 @@
 
 	if (!event) return FALSE;
 
-	if (event->button != 3)
-		return FALSE;
+    // Show context menu
+	//if (event->button != 3)
+	//	return FALSE;
 
 	if (!gtk_tree_selection_get_selected
 		(folderview->selection, NULL, &iter))
@@ -1879,6 +1880,17 @@
 			return TRUE;
 		}
 		break;
+    case GDK_Menu:
+        // Show context menu
+		if (folderview->selected) {
+			selected = gtk_tree_row_reference_get_path
+				(folderview->selected);
+            gtk_tree_selection_select_path(folderview->selection, selected);
+            folderview_menu_popup(folderview, event);
+            gtk_tree_path_free(selected);
+            return TRUE;
+		}
+        break;
 	default:
 		break;
 	}
